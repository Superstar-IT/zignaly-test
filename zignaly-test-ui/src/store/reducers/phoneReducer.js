import * as TYPES from "../types";

const initState = {
  phones: [],
  selectedPhone: null,
};
  
const phoneReducer = (state = initState, action) => {
  switch (action.type) {
    case TYPES.SET_PHONES:
      return { ...state, phones: action.phones };
    case TYPES.SELECT_PHONE:
      return { ...state, selectedPhone: action.phone };
    default:
      return { ...state };
  }
}

export { 
  initState,
  phoneReducer
};