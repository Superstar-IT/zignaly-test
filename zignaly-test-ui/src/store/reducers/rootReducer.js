import { combineReducers } from 'redux';
import { phoneReducer } from './phoneReducer';

const rootReducer = combineReducers({
  phones: phoneReducer,
});

export default rootReducer;