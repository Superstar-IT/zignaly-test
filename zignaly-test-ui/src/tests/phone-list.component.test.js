import React from "react";
import { Provider } from "react-redux";
import configureMockStore  from "redux-mock-store";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import PhoneListContainer from "../components/phone-list.component";
import { initState } from "../store/reducers/phoneReducer";

describe('PhoneDetail component', () => {
    const createMockStore = configureMockStore({ phones: initState });
    let mockStore;

    beforeEach(() => {
        mockStore = createMockStore({ phones: initState });
    })

    it('should redirect to /phones', () => {
        const phoneDetails = shallow(
            <Provider store={mockStore}>
                <PhoneListContainer />
            </Provider>)
            .dive()
        const instance = phoneDetails.instance();

        expect(toJson(phoneDetails)).toMatchSnapshot();
    });
});
