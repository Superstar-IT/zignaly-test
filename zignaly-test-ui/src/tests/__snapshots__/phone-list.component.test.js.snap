// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetail component should redirect to /phones 1`] = `<PhoneListContainer />`;
